### PR TITLE
Return sgkit sample_id in individual metadata

### LINKED ISSUE #871: Record sgkit sample_ids in individual metadata
As these are always present in sgkit, we should bring them through to the final TS. See https://github.com/benjeffery/tsinfer-snakemake/issues/17

### PR DESCRIPTION
**Pull request by benjeffery**
Fixes #871

### CODE REVIEW

**Comment by benjeffery** [comment]
Looks like we have some OSX regressions to do with the unwriteable file tests. Will fix in another PR.

File: tsinfer/formats.py [hunk 1]
```diff
@@ -2512,10 +2512,21 @@ def individuals_metadata_schema(self):
     @functools.cached_property
     def individuals_metadata(self):
         schema = tskit.MetadataSchema(self.populations_metadata_schema)
-        try:
-            return [schema.decode_row(r) for r in self.data["individuals_metadata"][:]]
-        except KeyError:
-            return [{} for _ in range(self.num_individuals)]
+        if "individuals_metadata" in self.data:
```
Comment (jeromekelleher) [inline tsinfer/formats.py:2515]: Took me a few minutes to figure this out - maybe a comment explaining that we always include the sgkit_sample_id, whether or not the user specified other metadata? What happens if the user already has ``sgkit_sample_id`` defined in their metadata?

This needs a CHANGELOG update too.

**Review by jeromekelleher — COMMENTED** [review]
LGTM, needs a bit of care with corner cases and docs though

File: tsinfer/formats.py (hunk 1 above)
Comment (benjeffery) [reply tsinfer/formats.py:2515]: I think if the key already exists that we silently don't set it. I don't think it is worth a warning as sgkit_sample_id is so specific it is likely to be the same string.

Do we need a changelog? None of the sgkit stuff is released yet.

Comment (jeromekelleher) [reply tsinfer/formats.py:2515]: True enough - I can see us forgetting to document this addition to the metadata though, so worth starting the CHANGELOG for sgkit stuff now?

**Comment by benjeffery** [comment]
Ok added a commit here, which prevents clobbering and tests that. Also a quick start at the sgkit changelog.

**Comment by benjeffery** [comment]
@mergifyio rebase

**Pull request merged**